File: 2021/05/28/main.py
class Solution:
    def maximumUniqueSubarray(self, nums: List[int]) -> int:
        l, s, r, i = set(), 0, 0, 0
        for n in nums:
            while n in l:
                s -= nums[i]
                l.remove(nums[i])
                i += 1
            l.add(n)
            s += n
            r = max(r, s)
        return r

File: 2021/05/28/main.go
package main

func maximumUniqueSubarray(nums []int) int {
	l, s, r, i := [10001]bool{}, 0, 0, 0
	for _, n := range nums {
		for l[n] {
			l[nums[i]] = false
			s -= nums[i]
			i++
		}
		l[n] = true
		s += n
		if s > r {
			r = s
		}
	}
	return r
}
